src/praisonai-agents/praisonaiagents/memory/memory.py

```
@@ -1150,8 +1153,19 @@ def reset_long_term(self):
                 self._log_verbose(f"Error clearing MongoDB long-term memory: {e}", logging.ERROR)
         if self.use_rag and hasattr(self, "chroma_client"):
-            self.chroma_client.reset()  # entire DB
-            self._init_chroma()         # re-init fresh
+            # Scope the reset to this instance's own collection. A full
+            # ``chroma_client.reset()`` wipes *every* collection in the shared
+            # persistent store, destroying other agents' long-term memory that
+            # happens to live in the same directory.
+            collection_name = getattr(self, "_collection_name", "memory_store")
+            try:
+                self.chroma_client.delete_collection(name=collection_name)
+            except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)